qutip/tests/solver/test_correlation.py

```diff
@@ -269,3 +267,37 @@ def test_correlation_2op_1t_known_cases(solver,
     cmp = qutip.correlation_2op_1t(H, state, times, c_ops, a_op, b_op, solver=solver)
 
     np.testing.assert_allclose(base, cmp, atol=0.25 if solver == 'mc' else 2e-5)
+
+
+def test_correlation_timedependant_op():
+    num = qutip.num(2)
+    a = qutip.destroy(2)
+    sx = qutip.sigmax()
+    sz = qutip.sigmaz()
+    times = np.arange(4)
+    # swith between sx and sz at t=1.5
+    A_op = qutip.QobjEvo([[sx, lambda t: t<=1.5], [sz, lambda t: t>1.5]])
+
+    cmp_sx = qutip.correlation_2op_1t(num, None, times, [a], sx, sx)
+    cmp_sz = qutip.correlation_2op_1t(num, None, times, [a], sz, sx)
+    cmp_switch = qutip.correlation_2op_1t(num, None, times, [a], A_op, sx)
+    np.testing.assert_allclose(cmp_sx[:2], cmp_switch[:2])
+    np.testing.assert_allclose(cmp_sz[-2:], cmp_switch[-2:])
+
+
+def test_alternative_solver():
+    from qutip.solver.mesolve import MeSolver
+    from qutip.solver.brmesolve import BRSolver
+
+    H = qutip.num(5)
+    a = qutip.destroy(5)
+    a_ops = [(a+a.dag(), qutip.coefficient(lambda _, w: w>0, args={"w":0}))]
+
+    br = BRSolver(H, a_ops)
+    me = MeSolver(H, [a])
```

Comment:
If we support `HEMOSolver` I guess we should also add it here.

Reply:
I do not know how to build a `Bath` that would represent a similar physic to compare with these results.
@hodgestar do you know any `HEOM` setup that can be compared to result of other solver or that we could compute analytically?
Otherwise I can add a test that `HEOMSolver` can run without error in correlation, but it could return junk and I would not know.

Reply:
Talking to Simon, `HEOMSolver` would not return the proper results since the density matrix does not contain the full state of the solver. It should be doable, but need it's own code. Removing for now.